external/fv3fit/tests/test_keras_testing.py

```diff
@@ -63,3 +71,101 @@ def test_dummy_model(dummy_model, gridded_dataset):
     )
 
     xr.testing.assert_allclose(ds_pred, ds_target)
+
+
+@pytest.mark.parametrize(
+    "input_variables, output_variables",
+    [
+        pytest.param(["feature0", "feature1"], ["pred0"], id="2_1"),
+        pytest.param(["feature0", "feature1"], ["pred0", "pred1"], id="2_2"),
+        pytest.param(["feature0"], ["pred0", "pred1"], id="1_2"),
+    ],
+)
+def test_constant_model_predict(input_variables, output_variables, nz):
+    predictor = fv3fit.testing.ConstantOutputPredictor(
+        sample_dim_name="sample",
+        input_variables=input_variables,
+        output_variables=output_variables,
+    )
+    gridded_dataset = get_gridded_dataset(nz)
+    outputs = get_first_columns(gridded_dataset, output_variables)
+    predictor.set_outputs(**outputs)
+    ds_stacked = safe.stack_once(
+        gridded_dataset, "sample", [dim for dim in gridded_dataset.dims if dim != "z"]
+    ).transpose("sample", "z")
+
+    ds_pred = predictor.predict(ds_stacked)
+
+    assert sorted(list(ds_pred.data_vars.keys())) == sorted(output_variables)
+    for name in output_variables:
+        assert np.all(ds_pred[name].values == outputs[name][None, :])
+
+
+@pytest.mark.parametrize(
+    "input_variables, output_variables",
+    [
+        pytest.param(["feature0", "feature1"], ["pred0"], id="2_1"),
+        pytest.param(["feature0", "feature1"], ["pred0", "pred1"], id="2_2"),
+        pytest.param(["feature0"], ["pred0", "pred1"], id="1_2"),
+    ],
+)
+def test_constant_model_predict_columnwise(input_variables, output_variables, nz):
```

Comment:
These three tests repeat a lot of code, could you make the predictor with set outputs as fixture?

Reply:
Avoiding fixtures for this to avoid too much magic coupling, but I reduced duplicated code a little bit using a helper function.